python/ray/data/tests/test_all_to_all.py

```diff
@@ -1730,7 +1730,7 @@ def test_random_shuffle_check_random(shutdown_only):
             prev = x
 
 
-def test_random_shuffle_with_custom_resource(ray_start_cluster):
+def test_random_shuffle_with_custom_resource(ray_start_cluster, use_push_based_shuffle):
```

Comment:
doesn't seem like `use_push_based_shuffle` is actually being used by the test?

Reply:
`use_push_based_shuffle` is actually a fixture that will set `ctx.use_push_based_shuffle` to True/False and run the test twice. The first time I saw this, I was confused as well. But we do have many such usages already.